include/settings/settings.h

```diff
@@ -160,16 +138,17 @@ int settings_save_one(const char *name, char *var);
  *
  * @param name Name/key of the settings item.
  * @param val_str Value of the settings item.
+ * @param len Length of value string.
  *
  * @return 0 on success, non-zero on failure.
  */
-int settings_set_value(char *name, char *val_str);
+int settings_set_value(char *name, void *val_str, size_t len);
```

Comment:
This looks very confusing to me. If you say "string" then it normally (at least in C) means a nul-terminated string, in which case a separate length is not needed. Also, if you say "string" the type should be `char *` and not `void *`. Could you clarify what's the idea here?

Reply:
I will change the nam as the value is byte array instead of string:
`int settings_set_value(char *name, void *val, size_t len);`

Reply:
Ok, that makes things clear. Note that `settings_save_one()` has the same issue.